src/libs/NetworkConnection.ts

```diff
@@ -112,6 +112,7 @@ function subscribeToNetInfo(): () => void {
             reachabilityUrl: `${CONFIG.EXPENSIFY.DEFAULT_API_ROOT}api/Ping?accountID=${accountID || 'unknown'}`,
             reachabilityMethod: 'GET',
             reachabilityTest: (response) => {
+                hasPendingNetworkCheck = false;
```

Comment:
I see this variable is used in `recheckNetworkConnection`... I _think_ that's just a manual call to check for the connection, is that correct?
And if so, why would we need to update this variable both here and in the current place we are updating it?

Reply:
Ah yeah you're right. I'll fix that since any check including NetInfo.refresh() will now check for the state first.

Reply:
Fixed this.

I kept
```
if (hasPendingNetworkCheck) {
        return;
    }
```
so that we only see the `[NetworkConnection] recheck NetInfo` Log when necessary. Otherwise, we would see the log every 60s. But I can remove it if we want to add the Log to `reachabilityShouldRun`. The only problem with this is we won't be able to differentiate whether it came from the automatic or manual recheck.